### ISSUE TITLE
Allow the accepted types to be specified in the resource

### ISSUE THREAD

**Issue by mvanholsteijn** [hashicorp/terraform-provider-http#15]
Hi there,
I would like to download the SAML metadata document and use it in my SAML provider in AWS. Unfortunately the http provider does not allow you to do that. 

My suggestion is to use the "Accept" request header value to determine whether or not the data is accepted.

### Terraform Version
```
Terraform v0.11.7
+ provider.auth0 v0.1.11
+ provider.http v1.0.1
```
### Affected Resource(s)

- http

### Terraform Configuration Files
```hcl
data "http" "auth0-saml-metadata" {
  url = "https://${var.auth0_domain}/samlp/metadata/${auth0_client.oauth-cli.client_id}"
		request_headers {
    "Accept" = "application/xml"
  }
}

resource "aws_iam_saml_provider" "default" {
  name                   = "auth0-${replace(var.auth0_domain,".","-")}-provider"
  saml_metadata_document = "${data.http.auth0-saml-metadata.body}"
}
```

### Expected Behavior
The SAML metadata document is downloaded and its body passed into the IAM SAML provider.

### Actual Behavior
```
* data.http.auth0-saml-metadata: data.http.auth0-saml-metadata: Content-Type is not a text type. Got: application/xml; charset=utf-8
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
```
data "http" "auth0-saml-metadata" {
  url = "http://httpbin.org/xml"
}
```

### References
- #13 
- #1

**Comment by tmccombs** [comment]
@mvanholsteijn are you planning on opening a PR to address this?

**Comment by mvanholsteijn** [comment]
@tmccombs You mean https://github.com/terraform-providers/terraform-provider-http/pull/16 ?

**Comment by paultyng** [comment]
This should be resolved now in #50
